instrumentation/opentelemetry-instrumentation-fastapi/src/opentelemetry/instrumentation/fastapi/__init__.py

```diff
@@ -54,7 +59,10 @@ def instrument_app(
             app.add_middleware(
                 OpenTelemetryMiddleware,
                 excluded_urls=excluded_urls,
-                span_details_callback=_get_route_details,
+                default_span_details=_get_route_details,
```

Comment:
I guess some instrumentations that use `asgi` need to do some preprocessing, which is why we need this `span_details_callback`?

Reply:
Yup your understanding is correct. I've tried to find some cleaner work around this, but there appears to be no other way as these instrumentations have the specific use case of all of the created spans (4 spans?) to use FastAPI/Starlette routes as the span name.